lib/fluent/plugin/out_bigquery.rb

```diff
@@ -2,374 +2,303 @@
 
 require 'fluent/plugin/bigquery/version'
 
-require 'fluent/mixin/config_placeholders'
-require 'fluent/mixin/plaintextformatter'
-
 require 'fluent/plugin/bigquery/schema'
 require 'fluent/plugin/bigquery/writer'
 
-## TODO: load implementation
-# require 'fluent/plugin/bigquery/load_request_body_wrapper'
-
 module Fluent
-  ### TODO: error classes for each api error responses
-  # class BigQueryAPIError < StandardError
-  # end
-
-  class BigQueryOutput < TimeSlicedOutput
-    Fluent::Plugin.register_output('bigquery', self)
-
-    # https://developers.google.com/bigquery/browser-tool-quickstart
-    # https://developers.google.com/bigquery/bigquery-api-quickstart
-
-    ### default for insert
-    def configure_for_insert(conf)
-      raise ConfigError unless conf["method"] != "load"
-
-      conf["buffer_type"]                = "lightening"  unless conf["buffer_type"]
-      conf["flush_interval"]             = 0.25          unless conf["flush_interval"]
-      conf["try_flush_interval"]         = 0.05          unless conf["try_flush_interval"]
-      conf["buffer_chunk_limit"]         = 1 * 1024 ** 2 unless conf["buffer_chunk_limit"] # 1MB
-      conf["buffer_queue_limit"]         = 1024          unless conf["buffer_queue_limit"]
-      conf["buffer_chunk_records_limit"] = 500           unless conf["buffer_chunk_records_limit"]
-    end
-
-    ### default for loads
-    def configure_for_load(conf)
-      raise ConfigError unless conf["method"] == "load"
+  module Plugin
+    class BigQueryOutput < Output
+      Fluent::Plugin.register_output('bigquery', self)
+
+      helpers :event_emitter
```

Comment:
If secondary output uses legacy compatibility layer or secondary output uses `event_emitter` helper, primary output needs event_emitter.

cf. fluent/plugin/output.rb:320- in fluentd-core (v0.14.7)
```
          @secondary.router = router if @secondary.has_router?

```

Because of it, I think that almost primary output needs event_emitter currently.